src/callgrind.rs

```diff
@@ -4,38 +4,62 @@ use std::io;
 
 use initialize::StackFrame;
 
+// Stats about the relationship between two functions, one of which
+// calls the other.
 #[derive(Debug)]
 struct Call {
+    // How many times has the outer function called the inner one?
     count: usize,
+
+    // How many samples were found inside the inner function
+    // (including all sub-functions), when called by the outer?
     inclusive: usize,
 }
 
+// Stats about a single function.
 #[derive(Debug, Default)]
 struct Location {
+    // How many samples were found inside this function only, not
+    // including calls to sub-functions?
     exclusive: usize,
+
+    // Data about the calls from this function to other functions.
     calls: HashMap<StackFrame, Call>,
 }
 
+// Stats about all functions found in our samples.
 #[derive(Default, Debug)]
 struct Locations(HashMap<StackFrame, Location>);
 
+// Information about a function currently on the stack.
 #[derive(Debug)]
 struct StackEntry {
     frame: StackFrame,
+
+    // How many samples were found inside this call only?
     exclusive: usize,
+
+    // How many samples were found in this call, and sub-calls?
     inclusive: usize,
 }
 
+// Tracks statistics about a program being sampled.
 #[derive(Default, Debug)]
 pub struct Stats {
-    // Stored with the root of the callgraph at the start.
+    // The current stack, along with tracking information.
+    // The root function is at element zero.
     stack: Vec<StackEntry>,
```

Comment:
is it true that we only store this stack to improve performance while calculating callgrind data?